sdk/samples/iot/hub/src/paho_iot_hub_pnp_component_example.c

```diff
@@ -0,0 +1,859 @@
+  }
+
+  az_span destination_span = az_span_init((uint8_t*)destination, destination_size);
+  az_span remainder = az_span_copy(destination_span, mqtt_url_prefix);
+  remainder = az_span_copy(remainder, az_span_slice(iot_hub, 0, iot_hub_length));
+  remainder = az_span_copy(remainder, mqtt_url_suffix);
+  az_span_copy_u8(remainder, null_terminator);
+
+  return AZ_OK;
+}
+
+// Read the user environment variables used to connect to IoT Hub
+static az_result read_configuration_and_init_client(void)
+{
+  az_span cert = AZ_SPAN_FROM_BUFFER(x509_cert_pem_file);
+  AZ_RETURN_IF_FAILED(
+      read_configuration_entry(ENV_DEVICE_X509_CERT_PEM_FILE, NULL, false, cert, &cert));
+
+  az_span trusted = AZ_SPAN_FROM_BUFFER(x509_trust_pem_file);
+  AZ_RETURN_IF_FAILED(
+      read_configuration_entry(ENV_DEVICE_X509_TRUST_PEM_FILE, "", false, trusted, &trusted));
+
+  az_span device_id_span = AZ_SPAN_FROM_BUFFER(device_id);
+  AZ_RETURN_IF_FAILED(
+      read_configuration_entry(ENV_DEVICE_ID, NULL, false, device_id_span, &device_id_span));
+
+  az_span iot_hub_hostname_span = AZ_SPAN_FROM_BUFFER(iot_hub_hostname);
+  AZ_RETURN_IF_FAILED(read_configuration_entry(
+      ENV_IOT_HUB_HOSTNAME, NULL, false, iot_hub_hostname_span, &iot_hub_hostname_span));
+
+  // Paho requires that the MQTT endpoint be of the form ssl://<HUB ENDPOINT>:8883
+  AZ_RETURN_IF_FAILED(
+      create_mqtt_endpoint(mqtt_endpoint, (int32_t)sizeof(mqtt_endpoint), iot_hub_hostname_span));
+
+  // Initialize the hub client with the hub host endpoint and the default connection options
+  az_iot_hub_client_options options = az_iot_hub_client_options_default();
+  options.model_id = model_id;
+  AZ_RETURN_IF_FAILED(az_iot_hub_client_init(
+      &client,
+      az_span_slice(iot_hub_hostname_span, 0, (int32_t)strlen(iot_hub_hostname)),
+      az_span_slice(device_id_span, 0, (int32_t)strlen(device_id)),
+      &options));
+
+  return AZ_OK;
+}
+
+static void send_device_info(void)
+{
+  // Get the device info in a JSON payload and the topic to which to send it
+  az_result result;
+  if ((result = sample_pnp_device_info_get_report_data(&client, &publish_message)) != AZ_OK)
+  {
+    LOG_ERROR("Could not get the device info data: error code = 0x%08x", result);
+    exit(result);
+  }
+
+  // Send the MQTT message to the endpoint
+  mqtt_publish_message(publish_message.topic, publish_message.out_payload_span, 0);
+}
+
+// Callback invoked by pnp functions each time it finds a property in the twin document
+static void sample_property_callback(
+    az_span component_name,
+    az_span property_name,
+    az_json_token* property_value,
+    int32_t version,
+    void* user_context_callback)
+{
+  (void)user_context_callback;
+  if (az_span_ptr(component_name) == NULL || az_span_size(component_name) == 0)
+  {
+    LOG_ERROR(
+        "Property=%.*s arrived for Control component itself.  This does not support\
+                writeable properties on it (all properties are on subcomponents)",
```

Comment:
fun fact. writeable can be written with or without an `e` :) Sub-components i agree

Reply:
Yep, I understand, your call. Looks like this gets discussed often: https://github.com/seqan/seqan3/pull/1933#discussion_r446508506

I brought that up because we have existing usages of `writable` in the SDK but no usage of `writeable`.
https://github.com/Azure/azure-sdk-for-c/search?q=writable&unscoped_q=writable

https://github.com/Azure/azure-sdk-for-c/blob/43e4de8dcac97b144c24e0693028c88a17b98b91/sdk/src/azure/platform/az_curl.c#L101